library/spdm_crypt_lib/libspdm_crypt_cert.c

```diff
@@ -813,13 +813,13 @@ static bool libspdm_verify_set_cert_leaf_cert_basic_constraints(
     const uint8_t *cert, size_t cert_size, uint8_t cert_model, bool need_basic_constraints)
 {
     bool status;
-    /* basic_constraints from certificate. */
-    uint8_t cert_basic_constraints[LIBSPDM_MAX_BASIC_CONSTRAINTS_CA_LEN];
+    /* basic_constraints from certificate. Add space for pathLen. */
+    uint8_t cert_basic_constraints[LIBSPDM_MAX_BASIC_CONSTRAINTS_CA_LEN + 10];
```

Comment:
How 10 is calculated? Is that enough?

Reply:
Both mbedtls and openssl seem to provide APIs to check if CA is true.
https://docs.openssl.org/3.0/man3/X509_check_ca/#copyright
https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/x509_crt.h#L920

Maybe we could invent another libspdm_x509 API to return if CA is true instead of parsing the basic constraints raw bytes?

Reply:
>How 10 is calculated? Is that enough?

It's probably more than enough, but let me check what the largest ASN.1 DER integer size is. I'll make a macro for it too.

Reply:
> Both mbedtls and openssl seem to provide APIs to check if CA is true. https://docs.openssl.org/3.0/man3/X509_check_ca/#copyright https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/x509_crt.h#L920
> 
> Maybe we could invent another libspdm_x509 API to return if CA is true instead of parsing the basic constraints raw bytes?

The benefit of having the logic in `libspdm/library` is that the Integrator doesn't need to provide new functions / functionality as the SPDM specification adds or removes certificate fields. But yes, maybe for libspdm 4.0 we could have a basic certificate parsing interface where the Integrator / cryptography library handles most of the logic, but then also have a catch-all interface where the Integrator provides the raw ASN.1 DER to libspdm.

Reply:
Thank Steven. Current change looks good and yes, I think the enhancement could wait to be implemented in 4.0